src/primitive/capsule.rs

```diff
@@ -45,17 +59,46 @@ where
     }
 }
 
+impl<'a, S> From<&'a Capsule<S>> for Aabb3<S>
```

Comment:
I don't think converting from reference is idiomatic. It's supposed to have `move` semantics, hence the auto-derive of `Into` trait.